### PR TITLE
hpke: Exposing KDF API

### PR DESCRIPTION
**Pull request by armfazh**

### CODE REVIEW

File: hpke/algs.go
Comment (cjpatton) [inline hpke/algs.go:55]: This seems like option (2.) from https://github.com/armfazh/circl/pull/8#discussion_r542501005.

I still don't think this is a good idea. The complexity it adds to the API isn't needed, since we currently have no HPKE KEMs that aren't suitable for sender authentication. I vote we hold off on this change until we actually need it. At the very least, let's rever this change in the current PR, since it's not relevant to the API changes for KDF. We can discuss it later on.

File: hpke/algs.go
Comment (cjpatton) [inline hpke/algs.go:140]:
```suggestion
// Extract derives a pseudorandom key from a high entropy, secret input and a salt.
```

File: hpke/algs.go
Comment (cjpatton) [inline hpke/algs.go:148]:
```suggestion
// Expand derives a variable length, pseudorandom string from a pseudorandom
// key and an information string. The maximum output length is 255*N bytes,
// where N is the size (in bytes) of the KDF's output.
```

Comment (cjpatton) [inline hpke/algs.go:148]: Also, document under which conditions this function panics.

File: hpke/algs.go
Comment (cjpatton) [inline hpke/algs.go:152]: "pseudo-random" -> "pseudorandom". (Here and elsewhere.)

File: hpke/hpke_test.go
Comment (cjpatton) [inline hpke/hpke_test.go:19]: nit: Remove space.

**Review by cjpatton — CHANGES_REQUESTED** [review]
Looks good, but let's revert the change to `func (k KEM) Scheme()` and consider it in a later PR.

File: hpke/algs.go
Comment (armfazh) [reply hpke/algs.go]: reverted

File: hpke/util.go
Comment (cjpatton) [inline hpke/util.go:69]:
```suggestion
// Params returns the codepoints for the algorithms comprising the suite.
```

File: hpke/vectors_test.go
Comment (cjpatton) [inline hpke/vectors_test.go:34]: How about
```
kemID := KEM(v.KemID)
test.CheckOk(kemID.IsValid(), "unknown KEM algorithm", t)
kem := kemID.Scheme()
```

**Review by cjpatton — APPROVED** [review]
LGTM modulo some minor things.

File: hpke/vectors_test.go
Comment (armfazh) [reply hpke/vectors_test.go:34]: good one

**Pull request merged**
